openid4vc-high-assurance-interoperability-profile-sd-jwt-vc-1_0.md

```diff
@@ -215,6 +221,57 @@ This is an example of a Wallet Instance Attestation:
     * In the `input-descriptors` object, `id`, `name`, `purpose`, `group`, `format`, and `constraints` properties MUST be supported. In the `constraints` object, `limit_disclosure`, and `fields` properties MUST be supported. In the `fields` object, `path` and `filter` properties MUST be supported. A `path` MUST contain exactly one entry with a static path to a certain claim. A `filter` MUST only contain `type` elements of value `string` and `const` elements.
     * In the `submission_requirements` object, `name`, `rule (`pick` only)`, `count`, `from` properties MUST be supported.
 
+# OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+* MUST support Annex A in [@!OIDF.OID4VP] that defines how to use OID4VP over the W3C Digital Credentials API.
+  * Annex A.3.1. in [@!OIDF.OID4VP] MUST NOT be used. OID4VP requests over the W3C Digital Credentials API MUST be signed.
+* Wallet Invocation is done via the W3C Digital Credentials API or an equivalent platform API. Custom URL schemes MUST NOT be used.
+* Response Mode MUST be `w3c_dc_api.jwt`. Encryption of the response is mandatory.
+* MUST support Transaction Data defined in Sections 5.4 and 7.4 of [@!OIDF.OID4VP].
+* The DQCL query and response as defined in Section 6 of [@!OIDF.OID4VP] MUST be used. Presentation Exchange as defined in Sections 5.4 and 5.5 of [@!OIDF.OID4VP] MUST NOT be used. Below is the list of features in the DQCL query and response that MUST be supported:
+  * tbd
+
+## ISO mdoc specific requirements for OpenID for Verifiable Presentations over W3C Digital Credentials API
+
+  * The Credential format identifier MUST be `mso_mdoc`.
+  * mdoc Credential Format specific DCQL parameters as defined in Section 6.4.2. of [@!OIDF.OID4VP] MUST be used.
+  * Verifier MAY request more than one credential in the same request.
+  * DeviceResponse in the VP Token MUST contain only one mdoc. Therefore, when multiple mdocs are being returned, each mdoc should be returned in a separate DeviceResponse, each matching to a respective DCQL query.
+  * Response encryption MUST be done as defined in Section B.4.3.3.2 in [@!ISO.18013-7].
```

Comment:
I don't know if I am allowed to copy 18013-7 text into here? @martijnharing do you know if we can take 18013-7 text and put it in OIDF spec?

Reply:
I think the mechanism specified in 18013-7 has a number of issues which makes it not directly applicable in the context of the browser api. For example, it is not bound to the origin.

Reply:
yeah, i think we probably need to update that a little bit, and find a way to includ (hash of) session transcript in JWE apu or apv to meet the following ISO requirements:

- mdoc authentication (device signature) must be bound to the origin, e.g. RP URL (If verifier’s value in the request is the different from the origin value passed from the platform, response must fail)
- Response encryption authentication must be bound to the origin, e.g. RP URL. 18013-5 uses `salt: SHA-256(SessionTranscriptBytes)` as key derivation input to derive session encryption keys

Reply:
- mdoc auth is a concept in ISO and essentially signs/macs over the SessionTranscript
- JARM response encryption is a concept in OID4VP
- ISO uses Session Encryption in -5 which in turn uses the SessionTranscript in the key derivation process.

1. We need to define SessionTranscript specifically for mdoc auth.
2. We need to define how apu/apv can be used to address the ISO requirement on "response encryption authentication must be bound to the origin". This mechanism should not be based on `SessionTranscript` since an mdoc can be returned in combination with an SD-JWT VC for example, so I'm not in favor of `salt:SHA-256(SessionTranscriptBytes)` for apu/apv. A general mechanism is needed. We could define how apu/apv values can be computed to be bound to the origin which would address ISO's requirement on binding the response encryption to the origin. This way, the mechanism can be used for mdocs, SD-JWT VCs, both, and other credential types.

Reply:
yes, I think you are right and I have not thought through enough. we should not design in a way that would imply that JWE needs to be done in a way that is different whether mdoc is being returned or not.

Reply:
WG discussion. 
agreement to encrypt the response. agreement to start with using JWE. agreement to have the same JWE requirements regardless of the credential format. discussion point is how to generate JWE. 

does not seem to be agreement to use JWE as defined in 18103-7 already. there seem to be some alignment on detached apu and apv being valuable?

option 1: improve how JWE with ECDH-ES is done. - change apu and apv values? 
option 2: define how to do JWE with HPKE. draft in IETF (https://datatracker.ietf.org/doc/html/draft-ietf-jose-hpke-encrypt-02) is not exactly stable. AAD can be set to the combination of [origin, clientId and nonce and encryption public key of the verifier].

Reply:
I suggest in the interests of making progress / merging this PR soon that we remove this sentence for now and have a separate discussion of the pros/cons of the different ways forward. For clarity, this results in the situation where encryption is mandatory, but how you encrypt is not defined.

```suggestion
```

Reply:
#131